cli/commands/find/command.go

```diff
@@ -0,0 +1,84 @@
+// Package find provides the ability to find Terragrunt configurations in your codebase
+// via the `terragrunt find` command.
+package find
+
+import (
+	"github.com/gruntwork-io/terragrunt/cli/flags"
+	"github.com/gruntwork-io/terragrunt/internal/cli"
+	"github.com/gruntwork-io/terragrunt/internal/errors"
+	"github.com/gruntwork-io/terragrunt/internal/experiment"
+	"github.com/gruntwork-io/terragrunt/options"
+)
+
+const (
+	CommandName  = "find"
+	CommandAlias = "fd"
+
+	FormatFlagName = "format"
+	JSONFlagName   = "json"
+	SortFlagName   = "sort"
+	HiddenFlagName = "hidden"
+)
+
+func NewFlags(opts *Options, prefix flags.Prefix) cli.Flags {
+	tgPrefix := prefix.Prepend(flags.TgPrefix)
+
+	return cli.Flags{
+		flags.NewFlag(&cli.GenericFlag[string]{
+			Name:        FormatFlagName,
+			EnvVars:     tgPrefix.EnvVars(FormatFlagName),
+			Destination: &opts.Format,
+			Usage:       "Output format for the find results. Valid values: text, json",
+			DefaultText: "text",
+		}),
+		flags.NewFlag(&cli.BoolFlag{
+			Name:        JSONFlagName,
+			EnvVars:     tgPrefix.EnvVars(JSONFlagName),
+			Destination: &opts.JSON,
+			Usage:       "Output in JSON format (equivalent to --format=json)",
+		}),
+		flags.NewFlag(&cli.GenericFlag[string]{
+			Name:        SortFlagName,
+			EnvVars:     tgPrefix.EnvVars(SortFlagName),
+			Destination: &opts.Sort,
+			Usage:       "Sort order for the find results. Valid values: alpha", // TODO: add dag
+			DefaultText: "alpha",
+		}),
+		flags.NewFlag(&cli.BoolFlag{
+			Name:        HiddenFlagName,
+			EnvVars:     tgPrefix.EnvVars(HiddenFlagName),
+			Destination: &opts.Hidden,
+			Usage:       "Include hidden directories in the find results.",
+		}),
+	}
+}
+
+func NewCommand(opts *options.TerragruntOptions) *cli.Command {
+	cmdOpts := NewOptions(opts)
```

Comment:
```suggestion
        prefix := flags.Prefix{CommandName}
	cmdOpts := NewOptions(opts)
```

Reply:
Why?

It's confusing to have environment variables sometimes pretended with the command name, and other times not (e.g. `TG_AUTH_PROVIDER_CMD` is not `TG_RUN_AUTH_PROVIDER_CMD`).

The majority of flags don't collide anymore.

Reply:
What I think, `run` flags shouldn't contain command name prefix. So `TG_AUTH_PROVIDER_CMD` should leave as it is `TG_AUTH_PROVIDER_CMD`. But other command flags should have the command prefix. Why?!

Environment variables can be set permanent in the shell. In these case some flags can collide, for example:

`hcl validate -json` TG_JSON
`find -json` TG_JSON

That's why I suggest that they be called that way

```
TG_FIND_JSON
TG_HCL_VALIDATE_JSON
```

Reply:
`TG_HIDDEN`

Also such short names of environment variables do not say much about what they refer to.

Reply:
The case from my task:

`terragurnt backend delete -bucket`

`TG_BUCKET` or `TG_BACKEND_DELETE_BUCKET`

Reply:
If we are consistent about following the rule that you can always set a flag by using `TG_<NAME_OF_FLAG>`, then it's easy to guess what every flag means. It means the same thing as the name of the flag. If users want, they can unset env vars or rely on CLI flags instead of env vars if they want to avoid collisions.

Whenever we can, we should design the interface so that nobody has to read the docs, or `--help` to know how to use it. That includes having to learn that there's a different naming convention for `run` (and `exec`) flags, and the rest.